### DISCUSSION TITLE
Fallback mechanism when property path is not found

### DISCUSSION THREAD

**Discussion by ddvlanck** [TREEcg/bucketizers#11] [Design decisions]
At this moment, whenever a bucketizer fails to extract a property path from an LDES member, it throws an error and the workflow it is being used in, stops working.

The desired behaviour should be that whenever the above situation happens, the LDES member is still assigned a bucket, but one that is different from the other buckets. The root file should then contain relations to the pages generated by the fallback mechanism.

Questions to be answered
- Naming convention for the pages generated by the fallback mechanism
- What if there is no root? (e.g. subject page bucketizer does not generate a root file)

**Comment by pietercolpaert** [comment]
My 2 cts:

### Naming convention

Should we prefix each bucket with the strategy so that buckets can’t conflict?

E.g.: 
```
root
substring/a
substring/aar
base/0
base/1
```
Or
```
geotiles/0/1/2
time/{starttimewindow} //time-based as a fallback for non-geospatial properties
base/0 //fall-back for when there’s no geospatial nor time-based property
```

### If there’s no root

I’d say there should be a function in the bucketizer that can indicate what the root should be? In a double linked list, any bucket can be assigned to be the root, so then it’s up to the bucketizer to randomly choose one?
